server/src/transport/http.rs

```diff
@@ -157,41 +158,47 @@ where
 {
 	let Batch { data, call } = b;
 
-	if let Ok(batch) = serde_json::from_slice::<Vec<Request>>(&data) {
-		let max_response_size = call.max_response_body_size;
-		let batch = batch.into_iter().map(|req| Ok((req, call.clone())));
-
-		let batch_stream = futures_util::stream::iter(batch);
-
-		let batch_response = batch_stream
-			.try_fold(
-				BatchResponseBuilder::new_with_limit(max_response_size as usize),
-				|batch_response, (req, call)| async move {
-					let response = execute_call(req, call).await;
-					batch_response.append(&response)
-				},
-			)
-			.await;
-
-		return match batch_response {
-			Ok(batch) => batch.finish(),
-			Err(batch_err) => batch_err,
-		};
-	}
+	if let Ok(batch) = serde_json::from_slice::<Vec<&JsonRawValue>>(&data) {
```

Comment:
parsing the request like this is slower than parsing directly `Vec<Request>` which is the most common usecase

overall, it looks to similar to master which I'm happy with considering that I changed to responses to be in order now.

Reply:
consider that my machine is a bit spurious for the benches:

```
group                                      batch-ordered                           master
-----                                      -------------                           ------
async/http_batch_requests/fast_call/10     1.26     57.9±8.16µs 168.6 KElem/sec    1.00     46.0±3.16µs 212.4 KElem/sec
async/http_batch_requests/fast_call/100    1.00   281.0±62.15µs 347.6 KElem/sec    1.20   337.5±31.42µs 289.4 KElem/sec
async/http_batch_requests/fast_call/2      1.32     43.6±9.62µs 44.8 KElem/sec     1.00     33.1±2.01µs 59.0 KElem/sec
async/http_batch_requests/fast_call/5      1.56    62.7±16.28µs 77.8 KElem/sec     1.00     40.2±2.96µs 121.5 KElem/sec
async/http_batch_requests/fast_call/50     1.04   182.8±44.29µs 267.2 KElem/sec    1.00   175.3±48.17µs 278.5 KElem/sec
async/ws_batch_requests/fast_call/10       1.00     82.2±9.43µs 118.8 KElem/sec    1.03     84.6±3.72µs 115.4 KElem/sec
async/ws_batch_requests/fast_call/100      1.00   305.9±37.64µs 319.3 KElem/sec    1.18   359.7±19.63µs 271.5 KElem/sec
async/ws_batch_requests/fast_call/2        1.00     42.1±3.17µs 46.4 KElem/sec     1.23     51.8±3.90µs 37.7 KElem/sec
async/ws_batch_requests/fast_call/5        1.00     30.9±1.19µs 158.0 KElem/sec    1.75    54.2±11.31µs 90.2 KElem/sec
async/ws_batch_requests/fast_call/50       1.00   185.8±22.98µs 262.9 KElem/sec    1.23   228.3±14.44µs 213.9 KElem/sec
sync/http_batch_requests/fast_call/10      1.00    72.9±12.58µs 134.0 KElem/sec    1.15     83.9±3.51µs 116.3 KElem/sec
sync/http_batch_requests/fast_call/100     1.14   335.2±42.17µs 291.4 KElem/sec    1.00   292.8±32.85µs 333.5 KElem/sec
sync/http_batch_requests/fast_call/2       1.28     45.0±9.56µs 43.4 KElem/sec     1.00     35.0±4.86µs 55.8 KElem/sec
sync/http_batch_requests/fast_call/5       1.15     49.3±9.73µs 99.1 KElem/sec     1.00     42.7±7.20µs 114.4 KElem/sec
sync/http_batch_requests/fast_call/50      1.00   150.9±41.80µs 323.6 KElem/sec    1.05    159.0±4.82µs 307.2 KElem/sec
sync/ws_batch_requests/fast_call/10        1.00    48.3±14.22µs 202.2 KElem/sec    1.27    61.4±22.71µs 159.1 KElem/sec
sync/ws_batch_requests/fast_call/100       1.00   272.4±46.48µs 358.5 KElem/sec    1.26   343.6±22.20µs 284.2 KElem/sec
sync/ws_batch_requests/fast_call/2         1.05     25.8±1.33µs 75.7 KElem/sec     1.00     24.5±0.72µs 79.6 KElem/sec
sync/ws_batch_requests/fast_call/5         1.04     30.9±1.28µs 158.3 KElem/sec    1.00     29.8±0.85µs 163.8 KElem/sec
sync/ws_batch_requests/fast_call/50        1.03   210.7±18.53µs 231.7 KElem/sec    1.00   205.2±10.41µs 237.9 KElem/sec

```